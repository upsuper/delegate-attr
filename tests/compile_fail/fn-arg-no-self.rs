use delegate_attr::delegate;

struct Foo(Vec<u8>);

#[delegate(self.0)]
impl Foo {
    fn len(x: &Foo) -> usize;
}

fn main() {}
