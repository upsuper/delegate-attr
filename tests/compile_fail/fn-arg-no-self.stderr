error: expected self
 --> $DIR/fn-arg-no-self.rs:7:12
  |
7 |     fn len(x: &Foo) -> usize;
  |            ^
